README.md

```diff
@@ -1,3 +1,15 @@
 # Path to Craftshmanship
 
 Ideas on getting to be a software craftsman. Practical lessons.
+
+# General Principles
+
+The following are principles aimed at harmonizing the documentation which is
+yet to be written.
+
+- always use the right terminology
+- understand how something works under the hood, but go at most 1 level deeper
+  at a time
+  For instance: when you learn a server-side web scripting language (Application
```

Comment:
Showing such dependencies is exactly what juniors search for. While we would express them, I believe we should come up with a practical path of getting there (where to learn it from, estimate how long it would take), not simply say "come back here once you know HTTP).